Cargo.toml

```diff
@@ -30,13 +30,14 @@ blake2b_simd = "1"
 ff = "0.11"
 fpe = "0.5"
 group = "0.11"
-halo2 = "=0.1.0-beta.1"
+halo2_gadgets = { version = "0.0", features = ["test-dependencies"] }
```

Comment:
Instead of always enabling this, do the same thing we do for `zcash_note_encryption`: have this dependency line in both `dependencies` and `dev-dependencies`, with the latter enabling this feature flag.